packages/core/src/RenderingEngine/helpers/getOrCreateCanvas.ts

```diff
@@ -54,5 +55,24 @@ export default function getOrCreateCanvas(
   const internalDiv =
     element.querySelector(viewportElement) || createViewportElement(element);
 
-  return internalDiv.querySelector(canvasSelector) || createCanvas(internalDiv);
+  const canvas = (internalDiv.querySelector(canvasSelector) ||
+    createCanvas(internalDiv)) as HTMLCanvasElement;
+  const rect = internalDiv.getBoundingClientRect();
```

Comment:
The new additions require a lot of documentation of why it is needed, can you please add that?